pkg/proctree/fork_processing.go

```diff
@@ -0,0 +1,155 @@
+package proctree
+
+import (
+	"github.com/RoaringBitmap/roaring"
+
+	"github.com/aquasecurity/tracee/pkg/utils/types"
+	"github.com/aquasecurity/tracee/signatures/helpers"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type threadIDs struct {
+	ProcessIDs
+	Tid int
+}
+
+// processForkEvent add new process to process tree if new process created, or update process threads if new thread
+// created. Because the fork at start is only a copy of the father, the important information regarding of the
+// process information and binary will be collected upon execve.
+func (tree *ProcessTree) processForkEvent(event *trace.Event) error {
+	err := tree.processDefaultEvent(event)
+	if err != nil {
+		return err
+	}
+
+	newProcessInHostIDs, err := parseForkInHostIDs(event)
+	if err != nil {
+		return err
+	}
+
+	if newProcessInHostIDs.Pid == newProcessInHostIDs.Tid {
+		err = tree.processMainThreadFork(event, newProcessInHostIDs)
```

Comment:
nit: drop the `err = ...`  in the clauses, change to a `return` and you can drop the else afterwards.